provider/aws/aws_discover.go

```diff
@@ -149,6 +149,7 @@ func (p *Provider) Addrs(args map[string]string, l *log.Logger) ([]string, error
 		l.Printf("[INFO] discover-aws: Using default credential chain")
 		cfg, err = config.LoadDefaultConfig(context.TODO(),
 			config.WithRegion(region), // Specify your region
+			config.WithUseDualStackEndpoint(aws.DualStackEndpointStateEnabled),
```

Comment:
Question: Why is dual stack limited to only the default credential chain?

Reply:
thanks. you're right, I have raised another commit to enable dual-stack in non-default flow as well.